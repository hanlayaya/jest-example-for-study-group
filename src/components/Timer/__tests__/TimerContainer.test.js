import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import MockDate from 'mockdate';
import storeFake from '../../../utils/testUtils';

describe('container <Timer />', () => {
  let subject;
  let Component;
  let TimerComponent;

  beforeEach(() => {
    jest.resetAllMocks();
    // Mocked Date
    MockDate.set(1501951314962);

    const store = storeFake({});
    const Timer = require('../Timer').default;
    const TimerContainer = require('../TimerContainer').default;

    subject = mount(
      <Provider store={store}>
        <TimerContainer />
      </Provider>
    );

    Component = subject.find(TimerContainer);
    TimerComponent = Component.find(Timer);
  });

  it('should be same as snapshot', () => {
    expect(TimerComponent.html()).toMatchSnapshot();
  });
});
