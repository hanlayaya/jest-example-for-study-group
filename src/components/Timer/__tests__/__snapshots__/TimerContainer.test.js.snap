// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container <Timer /> should be same as snapshot 1`] = `"<div><span><!-- react-text: 3 -->資料最後更新時間：<!-- /react-text --><!-- react-text: 4 -->Sun Aug 06 2017 00:41:54 GMT+0800 (CST)<!-- /react-text --></span><a>更新</a></div>"`;
